### PR TITLE
BGDIINF_SB-2209: Replacing poolmanager by multiprocessing

### PR DESCRIPTION
**Pull request by rebert**
.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -47,7 +49,7 @@
       zip_safe=False,
       test_suite='nose.collector',
       install_requires=install_requires,
-      python_requires='>2.6, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <4',
+      python_requires='>3.6, <4',
```
Comment (ltclm) [inline setup.py:52]: should be `>3.5` to be consistent with the list above.

File: tool_aws/s3/rm.py [hunk 2]
```diff
@@ -155,7 +155,8 @@ def createParser():
         action='store',
         type=bboxType,
         default=None,
-        help='a bounding box in lv95')
+        help='a bounding box in lv95. Only works in combination with  \
```
Comment (ltclm) [inline tool_aws/s3/rm.py:158]: is it this way round or the other?

**Review by ltclm — APPROVED** [review]
did you test it?
looks good to me, just a few questions.
thank you!

File: tool_aws/s3/rm.py (hunk 2 above)
Comment (rebert) [reply tool_aws/s3/rm.py:158]: Both are needed, if with bbox. 

And if only prefix it is misleading to put -i (as it has no affect)

**Comment by rebert** [comment]
@ltclm I tested it on my playground putting some dummy tiles to zoomlevel 12 in lv95 and lv03 (4 tiles). I tested -i, -hr and if the bbox works.

BBOX:
```for layer in "ch.test-1.bbox"; do                                               
    for year in current; do                                                     
        for cs in 21781 2056; do                                                
            for zoom in 12; do                                                  
                for x in {0..1}; do                                             
                    for y in {0..1}; do                                         
                        aws --profile playground s3 cp har.png s3://my-tiles/1.0.0/${layer}/default/${year}/${cs}/${zoom}/${x}/${y}.png
                    done                                                        
                done                                                            
            done                                                                
        done                                                                    
    done                                                                        
done                                                                            
                                                                                
for layer in "ch.test-1.bbox"; do                                               
    for year in current; do                                                     
        for cs in 3857; do                                                      
            for zoom in 1; do                                                   
                for x in {0..1}; do                                             
                    for y in {0..1}; do                                         
                        aws --profile playground s3 cp har.png s3://my-tiles/1.0.0/${layer}/default/${year}/${cs}/${zoom}/${x}/${y}.png
                    done                                                        
                done                                                            
            done                                                                
        done                                                                    
    done                                                                        
done   
```

For prefix:
```
for layer in "ch.test-1.testen" "ch.test-2.testen"; do                          
    for year in 20211231 20221231 current; do                                   
        for cs in 21781 2056 3857 4326; do                                      
            for x in {0..16}; do                                                
                for y in {0..16}; do                                            
                    for image in "0.png" "1.png"; do                            
                        aws --profile playground s3 cp har.png s3://my-tiles/1.0.0/${layer}/default/${year}/${cs}/${x}/${y}/${image}
                    done                                                        
                done                                                            
            done                                                                
        done                                                                    
    done                                                                        
done 
``` 



F.ex:
`s3rm --profile playground -b my-tiles -p /1.0.0/ch.test-1.bbox/default/current/2056/* -hr 750 --bbox 2500000,1250000,2550000,1300000 -i png`

or
`s3rm --profile playground -b my-tiles -p /1.0.0/ch.test-1.bbox/default/current/2056/* -hr 750 --bbox 2500000,1250000,2550000,1300000 -i jpeg`

...


            `

File: tool_aws/s3/rm.py [hunk 3]
```diff
@@ -360,19 +361,16 @@ def deleteWithPrefix(opts, S3Bucket, keys):
...
             if len(keys):
-                pm = PoolManager(numProcs=opts.nbThreads)
-                pm.imap_unordered(
-                    deleteKeys,
-                    keys,
-                    1,
-                    callback=callback)
+                with ProcessPoolExecutor(
+                        max_workers=opts.nbThreads) as executor:
+                    executor.map(deleteKeys, keys, timeout=3 * 60, chunksize=1)
```
Comment (ltshb) [inline tool_aws/s3/rm.py:373]: This won't work, the `executor` outside of the `with ProcessPoolExecutor...` will always be `None`

**Review by ltshb — CHANGES_REQUESTED** [review]

File: tool_aws/s3/rm.py (hunk 2 above)
Comment (rebert) [reply tool_aws/s3/rm.py:158]: no, I don't think so

File: tool_aws/s3/rm.py (hunk 3 above)
Comment (rebert) [reply tool_aws/s3/rm.py]: Thanks. I removed the condition

**Review by ltshb — APPROVED** [review]
I did not test the tool but I guess you did ?

**Pull request merged**